operations/host_provision.go

```diff
@@ -85,6 +83,11 @@ func hostProvision() cli.Command {
 	}
 }
 
+// makeHostProvisioningScriptFile creates the working directory with the host
+// provisioning script in it. Returns the
+// Note: we have to write the host provisioning script to a file instead of
+// running it directly like with 'sh -c "<script>"' because the script will exit
+// before it finishes executing on Windows.
```

Comment:
does Windows user data have a short timeout? How does it help to run it this way?

Reply:
It doesn't have any documented timeout. My best hypothesis on the premature exit is that `exec` could be implemented using `cmd.exe`, which has [a documented maximum character length limit, which is by default 2047](https://docs.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation) and the script is definitely longer than that.

Reply:
Do you mean because jasper executes the process with exec? It doesn't look like exec invokes the shell from [the comments](https://github.com/golang/go/blob/05ddb879c73bc0c84ed1a80864aaf27b4a3d032f/src/os/exec/exec.go#L9)...

Reply:
exec in Windows eventually boils down to a call to the Windows API for [CreateProcessW](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw).

Regardless, I have no proof of how Windows works, only empirical evidence to suggest that PowerShell behaves badly and this invocation of exec won't work unless you execute it as a file.